ops.go

```diff
@@ -77,19 +77,45 @@ func Tail[T any](num int, arr []T) []T {
 	return arr[len(arr)-num : len(arr)-1]
 }
 
-// Drop drops up to the first `num` elements.
+// Drop allocates a new slice, with the first `num` elements dropped.
 func Drop[T any](num int, arr []T) []T {
 	if len(arr) < num {
 		return []T{}
 	}
-	return arr[num:]
+
+	newLen := len(arr) - num
+	slice := make([]T, 0, newLen)
+	slice = append(slice, arr[num:]...)
+
+	return slice
 }
 
-// DropString drops up to the first `num` runes of a string.
+// DropString allocates a new string, with the first `num` bytes of a string dropped.
 func DropString(num int, what string) string {
 	if len(what) < num {
 		return ""
 	}
+
+	newLen := len(what) - num
+	buffer := make([]byte, 0, newLen)
+	buffer = append(buffer, what[num:]...)
+
+	return string(buffer)
+}
+
+// Skip skips the first `num` elements.
+func Skip[T any](num int, arr []T) []T {
```

Comment:
Good, so the underlying array stays the same but slice takes a part of it.

Reply:
Should that be made clear in the doc comment?

Reply:
yes